text/trace/0192-messaging-semantic-conventions-spec.md

```diff
@@ -0,0 +1,660 @@
+  +------------+                             +------------+
+```
+
+#### Single message producer, single message push-based consumer with manual settlement
+
+```
+  PRODUCER                                   CONSUMER
+
+  +------------+                             +------------------+
+  | Publish m1 | . . . . . . . . . . . . . . | Deliver m1       |
+  +------------+                             +-----+-----------++
+                                                   | Settle m1 |
+                                                   +-----------+
+```
+
+#### Single message producer, single message push-based consumer with auto-settlement
+
+```
+  PRODUCER                                   CONSUMER
+
+                                           +--------------------------+
+                                           | Ambient                  |
+  +------------+                           +-+------------+-----------+
+  | Publish m1 | . . . . . . . . . . . . . . | Deliver m1 |
+  +------------+                             +------------+-----------+
+                                                          | Settle m1 |
+                                                          +-----------+
+```
+
+#### Batch message producer with "Create" spans, single message pull-based consumer
+
+```
+  PRODUCER                                        CONSUMER
+
+  +-------------------------+                     +------------+
+  | Publish                 |         . . . . . . | Receive m1 |
+  +-+-----------+-----------+         .           +------------+
+    | Create m1 | . . . . . . . . . . .
+    +-----------+-----------+                     +------------+
+                | Create m2 | . . . . . . . . . . | Receive m2 |
+                +-----------+                     +------------+
+```
+
+#### Batch message producer, single message push-based consumer
+
+```
+  PRODUCER                                CONSUMER
+
+  +---------+                             +------------+
+  | Publish | . . . . . . . . . . . . . . | Deliver m1 |
+  +---------+                             +------------+
+        .                                 +------------+
+        . . . . . . . . . . . . . . . . . | Deliver m2 |
+                                          +------------+
+```
+
+#### Batch message producer with "Create" spans populated before publish, single message pull-based consumer
+
+```
+  PRODUCER                                      CONSUMER
+
+  +-----------------------------------+
+  | Ambient                           |
+  +-+-----------+---------------------+         +------------+
+    | Create m1 | . . . . . . . . . . . . . . . | Receive m1 |
+    +-----------+-----------+                   +------------+
+                | Create m2 | . . . . . . . .
+                +-----------+---------+     .   +------------+
+                            | Publish |     . . | Receive m2 |
+                            +---------+         +------------+
```

Comment:
We don't seem to be consistent within pull-based consumer scenarios in what the parent context should be for a receive span. I believe the intent is that the parent should be the application's ambient span?

Although intermediary instrumentation is out of scope, it's something I'm thinking about and there is a decision to make as to how the transport context of the message fits into these traces. If you care about the path the message takes, it's nice to have the receive span as a child of the message's context. But since it is the application's "pull" that caused the span, it's probably more correct for the receive span to be a child of the application's ambient span.

If we agree this is the right approach, it probably makes the most sense for the receive span to link to the message's transport context (although no need to mention here as that would be out of scope). If there is no ambient context, it seems as though it would be nicer to be a child rather than create a brand new trace with no parent at all. On the one hand, this means we have inconsistent structure. On the other hand, does splitting the traces up help with anything?